File: csharp/FantasyBattle/BasicBuff.cs
namespace FantasyBattle
{
    public class BasicBuff : Buff
    {
        public BasicBuff(float soakModifier, float damageModifier)
        {
            SoakModifier = soakModifier;
            DamageModifier = damageModifier;
        }

        public float SoakModifier { get; }
        public float DamageModifier { get; }
    }
}

File: cpp/model/BasicBuff.cpp
#include "BasicBuff.h"

BasicBuff::BasicBuff(float soakModifier, float damageModifier): soakModifier(soakModifier), damageModifier(damageModifier) {

}

float BasicBuff::getDamageModifier() {
    return damageModifier;
}

float BasicBuff::getSoakModifier() {
    return soakModifier;
}
